invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/LocationList/components/InternalLocationList/state/actions.js

```diff
@@ -24,3 +35,36 @@ export const fetchInternalLocations = () => {
       });
   };
 };
+
+export const deleteInternalLocation = ilocPid => {
+  return async dispatch => {
+    dispatch({
+      type: DELETE_IS_LOADING,
+    });
+
+    await internalLocationApi
+      .del(ilocPid)
+      .then(response => {
+        dispatch({
+          type: DELETE_SUCCESS,
+          payload: { internalLocationPid: ilocPid },
+        });
+        setTimeout(() => {
+          dispatch(fetchInternalLocations());
+          dispatch(
+            sendSuccessNotification(
```

Comment:
I think we should discussed this since it affects the UX and other places like the barcode checkout needs to use the same functionality. It might be a good idea to create a function that handles this that can be re-used.

One minor disadvantage of doing it this way is that the system might feel slower and clunkier even if it takes the same amount of time in the end. Getting that notification as soon as it's done then spinning and loading the new data feels more responsive to me. What do you think?

Reply:
After shortly talking with @zzacharo yesterday my preferred UX would be to dispatch the notification as soon as we receive the `DELETE_SUCCESS` and then keep loading as you stated till we render the updated results. In the meanwhile the user has enough time to read the notification and close it.